usr/src/lib/libzfs_core/common/libzfs_core.c

```diff
@@ -203,6 +204,23 @@ lzc_clone(const char *fsname, const char *origin,
 	return (error);
 }
 
+int
+lzc_promote(const char *fsname, const char *origin)
```

Comment:
Having to specify the origin snapshot is a bit unfortunate (the kernel could figure it out), so it's not great to encode that in the (hopefully stable) API.  If you wanted to fix that while you're here, it would be great.  But I'll go ahead and approve this as-is; we can always turn "origin" into an unused, ignored parameter in the future.